AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift

```diff
@@ -207,6 +210,47 @@ class SDCardMobileSessionsSavingService: SDCardMobileSessionssSaver {
         }
     }
     
+    private func addMeasurementLocation(deviceID: String, measurements: SDCardMeasurementsRow) -> CLLocationCoordinate2D? {
+
+        let isMini = deviceID.lowercased().contains(AirBeamDeviceType.airBeamMini.rawName)
+        
+        if isMini {
+            guard let sessionUUID else {
+                Log.error("[SD Sync] Session doesn't have sessionUUID")
+                fatalError()
+            }
+            do {
+                let session = try context.existingSession(uuid: sessionUUID)
+                let filteredStreams = session.allStreams.filter { MeasurementStreamEntity in
+                    !MeasurementStreamEntity.allMeasurements!.isEmpty
+                }
+                let latestMeasurements = filteredStreams.map({ $0.allMeasurements?.last })
+                let sortedMeasurements = latestMeasurements.sorted { first, second in
+                    guard let first, let second else { return false }
+                    return first.time > second.time
+                }
+                guard let latestMeasurement = sortedMeasurements.first else {
+                    /* We need to change it to current phone's location */
+                    return session.location!
+                }
+                guard let location = latestMeasurement?.location else {
+                    // ^
+                    return session.location!
+                }
+                return location
```

Comment:
This seems like pipe processing data. It could probably be written functionally, with .map{} etc. But if we don't want to do that, I'd probably at least extract this to a function and name it accordingly. Like... getLastSessionLocation maybe?